nxt_editor/integration/maya/run_maya_graph.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/python
+#
+# Open or reference "sub_graphs" graph in Maya or Standalone and pass an nxt
+# graph, mayapy.exe path and parameters in the "_maya_sub_graph" node, and
+# this will run that graph in maya standalone
+#
+# Shell Example:
+# open a shell. Navigate to your bin/mayapy.exe file in your installation of
+# maya. Then run: mayapy.exe path/to/this/run_maya_graph.py -g path/to/nxt_graph/you_want_to_run.nxt
+#
+
+# command line arguments
+import argparse
+
+# Initialize parser
+parser = argparse.ArgumentParser(description="This is a cli for running the standalone maya")
+
+# Adding optional argument
+parser.add_argument("-g", "--graph_path", help="The path to the nxt graph you want to run.")
+parser.add_argument(
+    "-p",
+    "--parameters",
+    help="""The parameters you want to pass to the graph. Parameters are a string representing a 
+    dictionary.e.g. {'path/to/node.attr':'value'}""",
+),
+parser.add_argument(
+    "-s", "--start_node", help="The path to the nxt nofe inI the graph you want to run."
+)
+
+# Read arguments from command line
+args = parser.parse_args()
+
+
+if __name__ == "__main__":
+    if args.graph_path:
+        # import maya standalone
+        from maya import standalone
+
+        # initialize maya standalone
+        standalone.initialize()
+
+        # in maya import the execute graph for nxt
+        from nxt import execute_graph
+
+        # make sure you can evaluate the parameters as a dict since it's passed as a string
+        parameters = {}
+        #  make sure you can evaluate the start node
+        start_node = None
+        # if there are parameters passed. We make sure they get passed to the graph
+        if args.parameters:
+            if isinstance(args.parameters, str):
+                parameters = eval(args.parameters)
```

Comment:
Are you expecting these to be a string of a dict _or_ a dict? If so, `json.loads` is much safer for handling the string.
However, the more correct way to handle multiple types in a single arg is to use the `type` kwarg in the argument itself and write a type validator like so:
```py

def dict_or_string(value):
    if isinstance(value, dict):
        return value
    try:
        # Try parsing as a dictionary
        parsed_dict = json.loads(value)
        if isinstance(parsed_dict, dict):
            return parsed_dict
    except ValueError:
        raise TypeError('Passed value must be of type dict or string!')

parser.add_argument(
    "-p",
    "--parameters",
    help="""The parameters you want to pass to the graph. Parameters are a string representing a 
    dictionary.e.g. {'path/to/node.attr':'value'}""",
    type=dict_or_string
)
```

Reply:
A string of a dict can be passed like so: `-p "{\"foo\": 123}"`. While uncommon I think we should account for it.
@MichaelAldrich mentioned the `--parameters` flag should support a filepath to a json parameters file. So the previous function would be more like:
```py
def dict_or_string(value):
    if isinstance(value, dict):
        return value
    try:
        # Try parsing as a dictionary
        parsed_dict = json.loads(value)
        if isinstance(parsed_dict, dict):
            return parsed_dict
    except ValueError:
        pass
    # Check if value string is a valid parameters file
    if os.path.isfile(value):
        try:
            with open(value, 'r') as fp:
                return json.load(fp)
        except json.JSONDecodeError:
            pass
    raise TypeError('Passed value must be of type dict, string of a dict, '
                    'or filepath to parameters file!')
```

Reply:
@ImLucasBrown do we still want the ability to just pass a dictionary in a string that doesn't require json syntax or are we saying it's either json, or a string-formatted dictionary and filepath that is a json file. Nothing else?

Reply:
@ImLucasBrown I have updated the code. I also updated the constants file via @MichaelAldrich a while back to have an `EDITOR_DIR` constant we can use.